File: app/app/src/main/java/com/resourceit/app/interfaces/APIService.java
package com.resourceit.app.interfaces;

import com.resourceit.app.models.LoginModel;
import com.resourceit.app.models.StatmentModel;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface APIService {

    @Headers({
            "Content-Type: application/x-www-form-urlencoded"
    })
    @GET("statements/{id}")
    Call<StatmentModel> GetStatments(@Path("id") String id);



    @Headers({
            "Content-Type: application/x-www-form-urlencoded"
    })
    @FormUrlEncoded
    @POST("login")
    Call<LoginModel> DoLogin(@Field("user") String user, @Field("password") String password);

}

File: app/src/main/java/com/example/otavioaugusto/testesantander/utils/APIService.kt
package com.example.otavioaugusto.testesantander.utils

import com.example.otavioaugusto.testesantander.model.StatementListItem
import com.example.otavioaugusto.testesantander.model.Statements
import com.example.otavioaugusto.testesantander.model.User
import com.example.otavioaugusto.testesantander.model.UserResponse
import retrofit2.Call
import retrofit2.http.*

interface APIService {

    @POST("api/login")
    @FormUrlEncoded
    fun login(@Field("user") user: String,
              @Field("password") password: String): Call<UserResponse>

    @GET("api/statements/{id}")
    fun getStatements(@Path("id") id:String): Call<Statements>
}
